builtin/providers/aws/resource_aws_elb.go

```diff
@@ -725,44 +725,61 @@ func resourceAwsElbUpdate(d *schema.ResourceData, meta interface{}) error {
 
 		removed := expandStringList(os.Difference(ns).List())
 		added := expandStringList(ns.Difference(os).List())
+		subnetconn := meta.(*AWSClient).ec2conn
 
-		if len(removed) > 0 {
-			detachOpts := &elb.DetachLoadBalancerFromSubnetsInput{
-				LoadBalancerName: aws.String(d.Id()),
-				Subnets:          removed,
+		// This section sorts the removed/added subnets based on availability zones
+		dup_az_count := 0
+		for i, r_subnet_id := range removed {
+			r_subnet, err := subnetconn.DescribeSubnets(&ec2.DescribeSubnetsInput{
+				SubnetIds: []*string{r_subnet_id},
+			})
+			if err != nil {
+				return err
+			}
+			for j, a_subnet_id := range added {
+				a_subnet, err := subnetconn.DescribeSubnets(&ec2.DescribeSubnetsInput{
```

Comment:
So, I could be missing something totally weird here, but couldn't we do something like:

```go
removedSubnets := map[string]string{}
addedSubnets := map[string]string{}
for _, subnet_id := range removed {
  subnet, err := subnetconn.DescribeSubnets(&ec2.DescribeSubnetsInput{
    SubnetIds: []*string{a_subnet_id},
  })
  if err != nil {
    return err
  }
  removedSubnets[subnet_id] = *subnet.Subnets[0].AvailabilityZone
}
for _, subnet_id := range added {
  subnet, err := subnetconn.DescribeSubnets(&ec2.DescribeSubnetsInput{
    SubnetIds: []*string{a_subnet_id},
  })
  if err != nil {
    return err
  }
  addedSubnets[subnet_id] = *subnet.Subnets[0].AvailabilityZone
}
```

Which would let you drop that inner API request, making fewer API requests.